errcheck/creator/creator.go

```diff
@@ -52,7 +70,35 @@ func Errcheck() checker.Creator {
 						}
 						return okgo.NewIssueFromLine(line, wd)
 					},
-				)), nil
+				),
+			)
+			return &cleanupChecker{c, cleanup}, nil
 		},
 	)
 }
+
+type cleanupChecker struct {
+	okgo.Checker
+	cleanup func()
+}
+
+func (c *cleanupChecker) Check(pkgPaths []string, projectDir string, stdout io.Writer) {
+	if c.cleanup != nil {
+		defer c.cleanup()
```

Comment:
Use the pattern described in `signals.ContextWithShutdown` (https://github.com/palantir/pkg/blob/master/signals/signals.go#L33) to ensure that cleanup function is run even if process is killed using ctrl-C.